<launch>
    <!-- Remapping -->
    <remap from="imu" to="sensors/imu"/>
    <remap from="fix" to="GPS/fix"/>
    <remap from="vel" to="GPS/vel"/>
    <remap from="GPS_coord" to="GPS/xy_coord"/>

    <!-- Motor command distributor -->
    <node pkg="motor_control" name="motor_control" type="motor_control_node.py" output="screen">
        <param name="sim" value="False"/>
        <param name='compass_offset' value='109.0'/>
    </node>

    <!-- Arduino + Servo + IMU -->
    <node pkg="rosserial_python" name="ros_arduino" type="serial_node.py">
        <param name="port" value="/dev/serial/by-id/usb-Arduino_Srl_Arduino_Uno_FFFFFFFFFFFF515B2503-if00"/>
    </node>

    

    <!-- ADCP -->
    <node pkg="adcp_reader" name="adcp_reader" type="adcp_reader_node.py" output="screen"/>

    <!--  GPS -->
    <!-- Should origin be sent as a param?? -->
    <node pkg="gps_reader" name="gps_reader" type="GPS_node.py" output="screen">
	<rosparam>
           /originLat: 0.0
           /originLon: 0.0
        </rosparam>
    </node>

    <node pkg="nmea_navsat_driver" name="navsat_driver" type="nmea_serial_driver" output="screen">
        <param name="port" value="/dev/serial/by-id/usb-Prolific_Technology_Inc._USB-Serial_Controller-if00-port0"/>
	      <param name="baud" value="19200"/>
    </node>


    <!-- Rosbridge for Remote Control -->
    <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch"/>

   <!-- LIDAR -->	
    <include file="$(find ouster_ros)/os1.launch"/>

    <!-- Rosbag Recording -->
    <node
	pkg="rosbag"
	name="record_odometry"
	type="record"
	args="/GPS/xy_coord /adcp/data /sensors/imu /GPS/fix /os1_cloud_node/points /os1_cloud_node/imu -o /media/nvidia/51E77A0927E4AE03/Data"
    /> 

</launch>
